plugin_dev/src/com/google/idea/blaze/plugin/sync/IntellijScalaPluginSyncPlugin.java

```diff
@@ -0,0 +1,102 @@
+import com.google.idea.blaze.base.model.BlazeProjectData;
+import com.google.idea.blaze.base.model.BlazeVersionData;
+import com.google.idea.blaze.base.model.primitives.LanguageClass;
+import com.google.idea.blaze.base.model.primitives.WorkspaceType;
+import com.google.idea.blaze.base.projectview.ProjectViewSet;
+import com.google.idea.blaze.base.scope.BlazeContext;
+import com.google.idea.blaze.base.sync.BlazeSyncPlugin;
+import com.google.idea.blaze.base.sync.SourceFolderProvider;
+import com.google.idea.blaze.java.sync.JavaLanguageLevelHelper;
+import com.google.idea.blaze.java.sync.model.BlazeJavaSyncData;
+import com.google.idea.blaze.java.sync.projectstructure.JavaSourceFolderProvider;
+import com.google.idea.common.util.Transactions;
+import com.intellij.openapi.module.ModuleType;
+import com.intellij.openapi.module.StdModuleTypes;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.LanguageLevelProjectExtension;
+import com.intellij.pom.java.LanguageLevel;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Development environment support for intellij plugin projects with Scala language
+ */
+public class IntellijScalaPluginSyncPlugin implements BlazeSyncPlugin {
+
+  @Override
+  public ImmutableList<WorkspaceType> getSupportedWorkspaceTypes() {
+    return ImmutableList.of(WorkspaceType.INTELLIJ_PLUGIN);
+  }
+
+  @Nullable
+  @Override
+  public ModuleType getWorkspaceModuleType(WorkspaceType workspaceType) {
+    if (workspaceType == WorkspaceType.INTELLIJ_PLUGIN) {
+      return StdModuleTypes.JAVA;
+    }
+    return null;
+  }
+
+  @Override
+  public Set<LanguageClass> getSupportedLanguagesInWorkspace(WorkspaceType workspaceType) {
+    if (workspaceType == WorkspaceType.INTELLIJ_PLUGIN) {
+      return ImmutableSet.of(LanguageClass.SCALA);
+    }
+    return ImmutableSet.of();
+  }
+
+  @Nullable
+  @Override
+  public SourceFolderProvider getSourceFolderProvider(BlazeProjectData projectData) {
+    if (!projectData
+        .getWorkspaceLanguageSettings()
+        .isWorkspaceType(WorkspaceType.INTELLIJ_PLUGIN)) {
+      return null;
+    }
+    return new JavaSourceFolderProvider(projectData.getSyncState().get(BlazeJavaSyncData.class));
+  }
+
+  @Override
+  public void updateProjectSdk(
+      Project project,
+      BlazeContext context,
+      ProjectViewSet projectViewSet,
+      BlazeVersionData blazeVersionData,
+      BlazeProjectData blazeProjectData) {
+    if (!blazeProjectData
+        .getWorkspaceLanguageSettings()
+        .isWorkspaceType(WorkspaceType.INTELLIJ_PLUGIN)) {
+      return;
+    }
+
+    LanguageLevel javaLanguageLevel =
+        JavaLanguageLevelHelper.getJavaLanguageLevel(
+            projectViewSet, blazeProjectData);
+
+    // Leave the SDK, but set the language level
+    Transactions.submitWriteActionTransactionAndWait(
+        () -> {
+          LanguageLevelProjectExtension ext = LanguageLevelProjectExtension.getInstance(project);
```

Comment:
Not 100% sure if this is required, but `BlazeGoSdkUpdater.java` modifies SDK in WriteAction context

Reply:
@tpasternak I just did the same impl as in [IntellijPluginSyncPlugin](https://github.com/bazelbuild/intellij/blob/master/plugin_dev/src/com/google/idea/blaze/plugin/sync/IntellijPluginSyncPlugin.java#L96). Should I change it, or keep consistent with the existing code?

Reply:
Ok, probably it's fine